src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncRewriter.AsyncIteratorRewriter.cs

```diff
@@ -175,9 +175,14 @@ protected override void GenerateConstructor()
                     bodyBuilder.Add(F.Assignment(F.InstanceField(initialThreadIdField), managedThreadId));
                 }
 
+                if (instanceIdField is not null &&
+                    F.WellKnownMethod(WellKnownMember.Microsoft_CodeAnalysis_Runtime_LocalStoreTracker__GetNewStateMachineInstanceId) is { } getId)
+                {
+                    bodyBuilder.Add(F.Assignment(F.InstanceField(instanceIdField), F.Call(receiver: null, getId)));
```

Comment:
>bodyBuilder [](http://example.com/codeflow?start=20&length=11)

Consider adding `// this.instanceIdField = LocalStoreTracker.GetNewStateMachineInstanceId();`